azure_sdk/build.sh

```diff
@@ -0,0 +1,103 @@
+#!/bin/bash
+
+set -e
+
+AZURE_VER="lts_01_2022"
+AZURE="azure-iot-sdk-c"
+
+PREFIX_AZURE="${TOPDIR}/phoenix-rtos-ports/azure_sdk"
+PREFIX_AZURE_RAW="${PREFIX_AZURE}/${AZURE}"
+PREFIX_AZURE_BUILD="${PREFIX_BUILD}/azure_sdk"
+PREFIX_AZURE_SRC="${PREFIX_AZURE_BUILD}/${AZURE}-${AZURE_VER}"
+PREFIX_AZURE_MARKERS="${PREFIX_AZURE_BUILD}/markers"
+PREFIX_AZURE_ROOT="${PREFIX_AZURE_BUILD}/root"
+PREFIX_AZURE_LIB="${PREFIX_AZURE_ROOT}/lib"
+PREFIX_AZURE_INC="${PREFIX_AZURE_ROOT}/include"
+PREFIX_AZURE_BIN="${PREFIX_AZURE_ROOT}/bin"
+PREFIX_AZURE_PATCHES="${PREFIX_AZURE}/patches"
+
+b_log "Building azure iot sdk"
+
+update_options=(--init)
+if [ "${LONG_TEST}" = "y" ]; then
+	update_options=("${update_options[@]}" --recursive)
+fi
+
+# Download in the specified version, it can't be done via wget, because tar.gz does not include submodules
+mkdir -p "${PREFIX_AZURE_BUILD}" "${PREFIX_AZURE_MARKERS}"
+if ! [ -d "${PREFIX_AZURE_RAW}" ]; then
+	git clone -b "${AZURE_VER}" https://github.com/Azure/azure-iot-sdk-c.git "${PREFIX_AZURE_RAW}"
+fi
+
+# Copy the repository, add version suffix and update submodules
+if ! [ -d "${PREFIX_AZURE_SRC}" ]; then
+	cp -r "${PREFIX_AZURE_RAW}" "$PREFIX_AZURE_BUILD"
+	mv "${PREFIX_AZURE_BUILD}/${AZURE}" "${PREFIX_AZURE_SRC}"
+	(cd "${PREFIX_AZURE_SRC}" && git submodule update "${update_options[@]}")
+fi
+
+# Apply patches
+for patchfile in "${PREFIX_AZURE_PATCHES}"/*.patch; do
+	if [ ! -f "${PREFIX_AZURE_MARKERS}/$(basename "$patchfile").applied" ]; then
+		echo "applying patch: $patchfile"
+		[[ "${LONG_TEST}" != "y" ]] && [[ $(basename "$patchfile") == "05_c-utility_utests.patch" ]] && continue
+		[[ "${TARGET}" != "armv7m7-imxrt106x"* ]] && [[ "$(basename "$patchfile")" == "04_armv7m7-imxrt106x.patch" ]] && continue
+		patch -d "${PREFIX_AZURE_BUILD}" -p0 -i "$patchfile" && touch "${PREFIX_AZURE_MARKERS}/$(basename "$patchfile").applied"
+	fi
+	# There are some cpp sources, which is currently not supported in Phoenix-RTOS
+	[ "${LONG_TEST}" = "y" ] && rm -rf "${PREFIX_AZURE_SRC}/c-utility/testtools/micromock"
+done
+
+# Set the entered connection string in sample's source (if specified)
+if [ "$AZURE_CONNECTION_STRING" ]; then
+	sed -i "s/\"\[device connection string\]\"/\"${AZURE_CONNECTION_STRING}\"/g" \
+	"${PREFIX_AZURE_SRC}/iothub_client/samples/iothub_ll_telemetry_sample/iothub_ll_telemetry_sample.c" \
+	&& echo "Connection string has been set properly."
+fi
```

Comment:
else? what happens? should we output some error/warning?

Reply:
I would prefer to leaving it without warning. It's additional utility to present/verify how azure sdk may work. I think the default use case is building azure sdk to use it in newly created applications.